File: 시저암호/java/20221021/src/Solution.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

class Solution {
    public String solution(String s, int n) {
        String answer = convert(s, n);
        return answer;
    }

    public String moveCharacter(String character, int times) {
        char lastCharacter = character.equals(character.toLowerCase())
                ? 'z'
                : 'Z';

        return character.charAt(0) + times > lastCharacter
                ? String.valueOf((char) (character.charAt(0) + times - 26))
                : String.valueOf((char) (character.charAt(0) + times));
    }

    public String convert(String string, int times) {
        Pattern pattern = Pattern.compile("\\w");
        Matcher matcher = pattern.matcher(string);

        return matcher.replaceAll(character -> moveCharacter(String.valueOf(character.group()), times));
    }
}

File: 시저암호/javascript/20221021/src/solution.js
function solution(s, n) {
  const answer = convert({ string: s, times: n });
  return answer;
}

export function moveCharacter({ character, times }) {
  const lastCharCode = character === character.toLowerCase()
    ? 'z'.charCodeAt(0)
    : 'Z'.charCodeAt(0);

  return lastCharCode > character.charCodeAt(0) + times
    ? String.fromCharCode(character.charCodeAt(0) + times)
    : String.fromCharCode(character.charCodeAt(0) + times - 26);
}

export default function convert({ string, times }) {
  return string.replaceAll(/\w/g, (character) => moveCharacter({ character, times }));
}
